src/data_with_data_attributes.coffee

```diff
@@ -0,0 +1,24 @@
+define [
+  'jquery',
+], ($) ->
+
+  (itemDataAttribute, sectionDataAttribute) ->
+    isSelect = (node) -> node.nodeName is 'SELECT'
+
+    isInput = (node) ->
+      isSelect(node) || node.nodeName is 'INPUT' || node.nodeName is 'TEXTAREA'
+
+    value: (node) ->
+      if isInput(node)
```

Comment:
What about using 'text = if' here. Then 'text' could be removed in each of the condition blocks.